src/library_egl.js

```diff
@@ -27,16 +49,69 @@ var LibraryEGL = {
         EGL.setErrorCode(0x3008 /* EGL_BAD_DISPLAY */);
         return 0;
       }
-      // TODO: read attribList.
       if ((!config || !config_size) && !numConfigs) {
         EGL.setErrorCode(0x300C /* EGL_BAD_PARAMETER */);
         return 0;
       }
-      if (numConfigs) {
-        {{{ makeSetValue('numConfigs', '0', '1', 'i32') }}}; // Total number of supported configs: 1.
+
+      // config variables to compare against
+      var alpha = 0;
+      var samples = 0;
+      var buffers = 0;
+      var depth = 0;
+      var stencil = 0;
+
+      if (attribList) {
+        while (1) {
+          var param = {{{ makeGetValue('attribList', '0', 'i32') }}};
+          if (param == 0x3038) { // EGL_NONE
+            break;
+          }
+
+          var value = {{{ makeGetValue('attribList', '4', 'i32') }}};
+          switch (param) {
+            case 0x3021: // EGL_ALPHA_SIZE
+              alpha = value;
```

Comment:
It would be good to recognize `EGL_DONT_CARE` here. With this code that would get treated as an invalid value that no config matches, returning 0 configurations if any `EGL_DONT_CARE`s are specified.